Bottom: 0e2929142ddefae8cfba8d7f895c45119b53aaf5
Top:    2a1711f61cc47eda8f5ea9874a25a1c204c2f27d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 20:29:30 +0100

Workaround crash where contour is passed as NULL to poly_PreContour






---

diff --git a/src/polygon.c b/src/polygon.c
index c697f3e..5c2d1f9 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
